webapp-django/crashstats/settings/base.py

```diff
@@ -643,11 +644,22 @@ def COMPRESS_JINJA2_GET_ENVIRONMENT():
 # then set this to False
 SESSION_COOKIE_SECURE = config('SESSION_COOKIE_SECURE', True, cast=bool)
 
-# To get your Sentry key, go to https://errormill.mozilla.org/
-RAVEN_CONFIG = {
-    'dsn': config('RAVEN_DSN', '')  # see https://errormill.mozilla.org/
-}
-
+# When socorro is installed (python setup.py install), it will create
+# a file in site-packages for socorro called "socorro/socorro_revision.txt".
+# If this socorro was installed like that, let's pick it up and use it.
+try:
+    SOCORRO_REVISION = resource_string('socorro', 'socorro_revision.txt')
+except IOError:
+    SOCORRO_REVISION = None
+
+# Raven sends errors to Sentry.
+# The release is optional.
+raven_dsn = config('RAVEN_DSN', '')
+if raven_dsn:
```

Comment:
This if statement is very important. I'm glad I tested it. What it means is that if you have...
```
RAVEN_CONFIG = {
    'dsn': '',
}
```
...you'd get this error:
```
Traceback (most recent call last):
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/raven/contrib/django/models.py", line 175, in sentry_exception_handler
    client.captureException(exc_info=sys.exc_info(), request=request)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/raven/contrib/django/models.py", line 54, in <lambda>
    __getattr__ = lambda x, o: getattr(get_client(), o)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/raven/contrib/django/models.py", line 155, in get_client
    instance = Client(**options)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/raven/base.py", line 144, in __init__
    self.set_dsn(dsn, transport)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/raven/base.py", line 199, in set_dsn
    transport_registry=self._registry,
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/raven/conf/remote.py", line 75, in from_string
    raise InvalidDsn(ERR_UNKNOWN_SCHEME.format(url.scheme, value))
InvalidDsn: Unsupported Sentry DSN scheme:  ()
```